server/src/OnDeviceComponent.ts

```diff
@@ -148,26 +103,34 @@ export class OnDeviceComponent {
 		return this.handshakePromise;
 	}
 
-	private async sendRequest(request: OnDeviceComponentRequest, timeoutMilliseconds: number = 5000) {
+	// In some cases it makes sense to break out the last key path part as `field` to simplify code on the device
+	private breakOutFieldFromKeyPath(args: ODCCallFuncArgs | ODCObserveFieldArgs | ODCSetValueAtKeyPathArgs) {
+		const keyPathParts = args.keyPath.split('.');
+		return {...args, field: keyPathParts.pop(), keyPath: keyPathParts.join('.')};
+	}
+
+	private async sendRequest(type: ODCRequestTypes, args: ODCRequestArgs, options?: ODCRequestOptions) {
 		if (this.handshakeStatus !== 'complete') {
 			if (this.handshakeStatus === 'failed') {
 				throw new Error('Can not continue as handshake was not successful');
 			}
 			await this.sendHandShakeRequest();
 		}
-		return await this.sendRequestCore(request, timeoutMilliseconds);
+		return await this.sendRequestCore(type, args, options);
 	}
 
-	private async sendRequestCore(request: OnDeviceComponentRequest, timeoutMilliseconds: number = 5000) {
+	private async sendRequestCore(type: ODCRequestTypes, args: ODCRequestArgs, options?: ODCRequestOptions) {
 		await this.startServer();
 
 		const requestId = utils.randomStringGenerator();
-		const formattedRequest = {
+		const request: ODCRequest = {
 			id: requestId,
-			callbackPort: this.callbackListenPort,
-			type: request.type,
-			args: request.args
+			callbackPort: this.callbackListenPort!,
```

Comment:
very minor do we have a default value for callbackListenPort that we configure, may be we can make it mandatory if we always require it to be passed

Reply:
At this point it is guaranteed to be set hence why I force it